Tests/DatadogTests/Datadog/RUM/Instrumentation/Views/RUMViewsHandlerTests.swift

```diff
@@ -7,21 +7,238 @@
+        handler.notify_viewDidAppear(viewController: view2, animated: .mockAny())
+        handler.notify_viewDidDisappear(viewController: view2, animated: .mockAny())
+
+        // Then
+        XCTAssertEqual(commandSubscriber.receivedCommands.count, 0)
+    }
+
+    func testGivenViewControllerStarted_whenAppStateChanges_itStopsAndRestartsRUMView() throws {
+        let viewName: String = .mockRandom()
+        let viewControllerClassName: String = .mockRandom()
+        let view = createMockView(viewControllerClassName: viewControllerClassName)
+
+        // Given
+        predicate.result = .init(name: viewName, attributes: ["foo": "bar"])
+        handler.notify_viewDidAppear(viewController: view, animated: .mockAny())
+
+        // When
+        notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
+        dateProvider.advance(bySeconds: 1)
+        notificationCenter.post(name: UIApplication.willEnterForegroundNotification, object: nil)
+
+        // Then
+        XCTAssertEqual(commandSubscriber.receivedCommands.count, 3)
+
+        let stopCommand = try XCTUnwrap(commandSubscriber.receivedCommands[1] as? RUMStopViewCommand)
+        XCTAssertTrue(stopCommand.identity.equals(view))
+        XCTAssertEqual(stopCommand.attributes.count, 0)
+        XCTAssertEqual(stopCommand.time, .mockDecember15th2019At10AMUTC())
+
+        let startCommand = try XCTUnwrap(commandSubscriber.receivedCommands[2] as? RUMStartViewCommand)
+        XCTAssertTrue(startCommand.identity.equals(view))
+        XCTAssertEqual(startCommand.path, viewControllerClassName)
+        XCTAssertEqual(startCommand.name, viewName)
+        XCTAssertEqual(startCommand.attributes as? [String: String], ["foo": "bar"])
+        XCTAssertEqual(startCommand.time, .mockDecember15th2019At10AMUTC() + 1)
+    }
+
+    func testGivenViewControllerDidNotStart_whenAppStateChanges_itDoesNothing() throws {
+        let view = createMockViewInWindow()
+        let viewName: String = .mockRandom()
+
+        // Given
+        predicate.result = .init(name: viewName, attributes: ["foo": "bar"])
+        handler.notify_viewDidDisappear(viewController: view, animated: .mockAny())
+
+        // When
+        notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
+        dateProvider.advance(bySeconds: 1)
+        notificationCenter.post(name: UIApplication.willEnterForegroundNotification, object: nil)
+
+        // Then
+        XCTAssertEqual(commandSubscriber.receivedCommands.count, 0)
+    }
+
+    // MARK: - Interacting with predicate
+
+    func testGivenAppearedView_whenTransitioningBackAndForthFromThisViewToAnother_thenPredicateIsCalledOnlyTwice() {
+        class Predicate: UIKitRUMViewsPredicate {
+            var numberOfCalls = 0
+
+            func rumView(for viewController: UIViewController) -> RUMView? {
+                numberOfCalls += 1
+                return .init(name: .mockRandom())
```

Comment:
It differs from the [original test implementation](https://github.com/DataDog/dd-sdk-ios/blob/master/Tests/DatadogTests/Datadog/RUM/AutoInstrumentation/Views/UIKitRUMViewsHandlerTests.swift#L178): If the predicate rejects the vc, it is not added to the stack.

Reply:
Nice 👌, this clearly shows the improvement over previous approach, where UIKit handler would be invoked 5 times for this test 👍.